Composer/packages/client/src/components/CreationFlow/v2/DefineConversation.tsx

```diff
@@ -0,0 +1,369 @@
+
+import {
+  DialogCreationCopy,
+  QnABotTemplateId,
+  nameRegexV2,
+  invalidNameCharRegex,
+  mockLanguageOptions,
+  runtimeOptions,
+  defaultPrimaryLanguage,
+  defaultRuntime,
+} from '../../../constants';
+import { FieldConfig, useForm } from '../../../hooks/useForm';
+import { StorageFolder } from '../../../recoilModel/types';
+import { createNotification } from '../../../recoilModel/dispatchers/notification';
+import { ImportSuccessNotificationWrapper } from '../../ImportModal/ImportSuccessNotification';
+import { dispatcherState } from '../../../recoilModel';
+import { LocationSelectContent } from '../LocationSelectContent';
+
+// -------------------- Styles -------------------- //
+
+const textFieldlabel = {
+  label: {
+    root: [
+      {
+        fontWeight: FontWeights.semibold,
+      },
+    ],
+  },
+};
+
+const name = {
+  subComponentStyles: textFieldlabel,
+};
+
+const description = {
+  subComponentStyles: textFieldlabel,
+};
+
+const halfstack = {
+  root: [
+    {
+      flexBasis: '50%',
+    },
+  ],
+};
+
+const stackinput = {
+  root: [
+    {
+      marginBottom: '1rem',
+    },
+  ],
+};
+
+// -------------------- DefineConversation -------------------- //
+
+const MAXTRYTIMES = 10000;
+
+interface DefineConversationFormData {
+  name: string;
+  description: string;
+  schemaUrl: string;
+  primaryLanguage: string;
+  runtimeChoice: string;
+  location?: string;
+
+  templateDir?: string; // location of the imported template
+  eTag?: string; // e tag used for content sync between composer and imported bot content
+  urlSuffix?: string; // url to deep link to after creation
+  alias?: string; // identifier that is used to track bots between imports
+  preserveRoot?: boolean; // identifier that is used to determine ay project file renames upon creation
+}
+
+interface DefineConversationProps
+  extends RouteComponentProps<{
+    templateId: string;
+    location: string;
+  }> {
+  createFolder: (path: string, name: string) => void;
+  updateFolder: (path: string, oldName: string, newName: string) => void;
+  onSubmit: (formData: DefineConversationFormData, templateId: string) => void;
+  onDismiss: () => void;
+  onCurrentPathUpdate: (newPath?: string, storageId?: string) => void;
+  onGetErrorMessage?: (text: string) => void;
+  focusedStorageFolder: StorageFolder;
+}
+
+const DefineConversationV2: React.FC<DefineConversationProps> = (props) => {
+  const {
+    onSubmit,
+    onDismiss,
+    onCurrentPathUpdate,
+    templateId,
+    focusedStorageFolder,
+    createFolder,
+    updateFolder,
+  } = props;
+  const files = focusedStorageFolder?.children ?? [];
+  const writable = focusedStorageFolder.writable;
+
+  // template ID is populated by npm package name which needs to be formatted
+  const normalizeTemplateId = (templateId?: string) => {
+    if (templateId) {
+      return templateId.replace(invalidNameCharRegex, '_');
+    }
+  };
+
+  const getDefaultName = () => {
```

Comment:
was this function copied from somewhere? it seems out of place

Reply:
Yes this is from the original DefineConversation.tsx and slightly altered to provide a valid default project name to prepopulate the name field in the UI. There is a good bit of redundant code between the two flows which is intended since they will only live side by side for a short period of time.